tests/services/analysis_starter/test_balsamic_configurator.py

```diff
@@ -0,0 +1,439 @@
+        return_value=create_autospec(BedVersion, filename="NACG_goose_1_99_max_fluff.bed")
+    )
+
+    # WHEN resolving the bed file with an override flag
+    provided_bed_name = Path("NACG_goose")
+    bed_file = balsamic_configurator._resolve_bed_file(
+        case_missing_bed, panel_bed=provided_bed_name
+    )
+
+    # THEN the overridden bed file should be returned
+    assert bed_file == Path(bed_directory, "NACG_goose_1_99_max_fluff.bed")
+
+    # THEN the LIMS API should not be called
+    balsamic_configurator.lims_api.capture_kit.assert_not_called()
+
+
+def test__get_normal_sample_id():
+    """ "Tests that a case containing a normal sample returns the correct sample id."""
+    # GIVEN a case with a normal sample
+    case_with_normal = create_autospec(Case)
+    normal_sample = create_autospec(Sample, internal_id="normal_sample", is_tumour=False)
+    case_with_normal.samples = [normal_sample]
+
+    # WHEN getting the normal sample id
+    normal_sample_id = BalsamicConfigurator._get_normal_sample_id(case_with_normal)
+
+    # THEN the correct normal sample id should be returned
+    assert normal_sample_id == "normal_sample"
+
+
+def test__get_normal_sample_id_no_normal_sample():
+    """Tests that a case without a normal returns None."""
+    # GIVEN a case without a normal sample
+    case_with_tumor = create_autospec(Case)
+    tumour_sample = create_autospec(Sample, internal_id="tumour_sample", is_tumour=True)
+    case_with_tumor.samples = [tumour_sample]
+
+    # WHEN getting the normal sample id
+    normal_sample_id = BalsamicConfigurator._get_normal_sample_id(case_with_tumor)
+
+    # THEN the normal sample id should be None
+    assert normal_sample_id is None
+
+
+def test__get_tumor_sample_id():
+    """Tests that a case containing a tumor sample returns the correct sample id."""
+    # GIVEN a case with a tumor sample
+    case_with_tumor = create_autospec(Case)
+    tumor_sample = create_autospec(Sample, internal_id="tumor_sample", is_tumour=True)
+    case_with_tumor.samples = [tumor_sample]
+
+    # WHEN getting the tumor sample id
+    tumor_sample_id = BalsamicConfigurator._get_tumor_sample_id(case_with_tumor)
+
+    # THEN the correct tumor sample id should be returned
+    assert tumor_sample_id == "tumor_sample"
+
+
+def test__get_tumor_sample_id_no_tumor_sample():
+    """Tests that a case without tumor samples raises an error."""
+    # GIVEN a case without a tumor sample
+    case_with_normal = create_autospec(Case)
+    normal_sample = create_autospec(Sample, internal_id="normal_sample", is_tumour=False)
+    case_with_normal.samples = [normal_sample]
+
+    # WHEN getting the tumor sample id
+    # THEN it should raise a ValueError
+    with pytest.raises(BalsamicMissingTumorError):
+        BalsamicConfigurator._get_tumor_sample_id(case_with_normal)
+
+
+def test__build_cli_input_wgs_tumor_only(balsamic_configurator):
+    # GIVEN a case with one WGS sample
```

Comment:
```suggestion
    # GIVEN a case with one tumor WGS sample
```

Reply:
fixed in 55a9f5ae2a98a97a78b86eee6f8e13194d48b10d